### DISCUSSION TITLE
bootstrap:27 Uncaught TypeError: __webpack_require__.nmd is not a function

### DISCUSSION THREAD

**Discussion by mpacierz** [webpack/webpack#18306] [Q&A] [answered]
I saw a similar problem [here](https://github.com/webpack/webpack/discussions/14064?sort=top), but I can't solve it

My package.json webpack versions:
    "webpack": "5.89.0",
    "webpack-cli": "^5.1.4",
    "webpack-dev-server": "4.15.2"

Webpack.config:

```

	output: {
		path: outPath,
		publicPath: "/",
		filename: isProduction ? "[name].[contenthash].js" : "[name].[fullhash].js",
		chunkFilename: isProduction ?  "[name].[contenthash].js" : "[name].[fullhash].js",
		pathinfo: false,
		clean: true
	},

// static assets
			{
				test: /\.(png|jpg|gif|eot|ttf|woff|woff2)$/,
				type: "asset",
				parser: {
					dataUrlCondition: {
						maxSize: 8192 // Limit dla konwersji do base64
					}
				},
				generator: {
					filename: '[name].[hash][ext]'
				}
			},
			{
				test: /\.svg$/,
				issuer: /\.(ts)x?$/,
				use: ["@svgr/webpack"]
			},
			{
				test: /\.svg$/,
				issuer: /\.(sa|sc|c)ss$/,
				type: "asset/resource",
				generator: {
					filename: '[name].[hash].[ext]'
				}
			}

                ...
	devServer: {
			host: "0.0.0.0",
			port: 3000,
			open: false,
		//	hot: true,
			historyApiFallback: {
				disableDotRule: true
			}
	},

		]
```

I throw away CleanWebpackPlugin.

where can I look for errors?

**Comment by alexander-akait** [comment]
Please create reproducible test repo

**Reply by mpacierz** [reply]
I can't reproduce this error, but maybe full webpack configuration will help somehow
[link](https://codesandbox.io/p/sandbox/nmd-ngqvrx?file=%2Fpublic%2Findex.html%3A66%2C1)

**Reply by alexander-akait** [reply]
Can you provide steps to reproduce? I don't see any problems with your configuration...

**Reply by mpacierz** [reply]
I'll try to reproduce this error on sandbox for tomorrow, thank you for review

**Reply by mpacierz** [reply]
damn, still cant reproduce. Full error in console  
```
function
    at ../node_modules/react/cjs/react.development.js (react-jsx-runtime.development.js:1314:1)
    at __webpack_require__ (bootstrap:22:1)
    at fn (hot module replacement:61:1)
    at ../node_modules/react/index.js (index.js:6:1)
    at __webpack_require__ (bootstrap:22:1)
    at fn (hot module replacement:61:1)
```

I read that hot module replacement was deprecated (for my react version) in favor of react-refresh. 
Maybe improving the plugin will solve this problem. (just in case I don't have babel.config.js in my project)

**Reply by alexander-akait** [reply]
Yeah, try to replace it and let's look, feel free to feedback

**Reply by mpacierz** [reply, marked as answer]
Closed! Replacing react-hot-loader with
```
"react-refresh": "0.14.0",
"@pmmmwh/react-refresh-webpack-plugin": "0.5.11",
```
fixed the error.
